### PR TITLE
Ian/docker

### PR DESCRIPTION
**Pull request by Bellardia**
Introduces sweeping changes to support dockerized environments.

- Disable network optimizations in docker, as docker uses the host network controller.
- Disable NTP, as the container uses the host clock.
- Add ohai recipe for container detection, and resulting `node[:container]` attribute
- Disable process priority for php7.4-fpm in containerized environments, as the operation is not permitted.
- Disable systemd-resolved, DNS works fine with the host controller/dnsmasq
- Upgrade to timescaledb 3.7.2 / postgres 14
- Include intermediate certificates

### CODE REVIEW

File: base/recipes/gearman.rb [hunk 1]
```diff
@@ -88,6 +90,7 @@
 # Memcached seems to be sideloaded by gearrman; disable it
 service "memcached" do
   supports status: true, restart: true, reload: true, stop: true
+  provider Chef::Provider::Service::Systemd
```
Comment (ahosking) [inline base/recipes/gearman.rb:93]: This is clutch, thanks for finding/including this!

File: configure/files/container.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+Ohai.plugin(:Container) do
+  provides "container"
+
+  collect_data(:default) do
+    if File.exist?("/.dockerenv") || !!(File.read("/proc/1/cgroup") =~ %r[^\d+:\w+:/(lxc|docker)/])
+      container(true)
+    end
+  end
+end
```
Comment (ahosking) [inline configure/files/container.rb:9]: This is much better than the check I had in place. It is also a great example of an Ohai plugin I'll look to reference in the future. 👏

**Review by ahosking — APPROVED** [review]
Great work @Bellardia !
I've got some notes for future improvement but, thanks for picking this up and getting it across the line.

File: base/recipes/gearman.rb (hunk 1 above)
Comment (Bellardia) [reply base/recipes/gearman.rb:93]: that doesn't actually do anything btw, just included for consistency

**Pull request merged**